priorityq.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import random
+
+class Item(object):
+
+    def __init__(self, value, priority):
+        self.value = value
+        self.priority = priority
+
+    # Priority with lower values are higher priority than those
+    # with higher values. E.g. Item with priority 1 is higher
+    # t
+    def higher_pri_than(self, item):
+        return self.priority < item.priority
+
+
+class PriorityQ(object):
+
+    def __init__(self):
+        self.pqheap = BinHeap()
+
+    def insert(self, item):
+        self.pqheap.push(item)
+
+    def pop(self):
+        return self.pqheap.pop()
+
+    def peek(self):
+        return self.pqheap.peek()
+
+class BinHeap(object):
+
+    def __init__(self, iterable=None):
+        # heap will start at index 1 of heapList.
+        self.heapList = [0]
+        self._size = 0
+
+        # builds a heap from an iterable
+        if iterable is not None:
+            index = len(iterable) // 2
+            self._size = len(iterable)
+            self.heapList = self.heapList + list(iterable[:])
+            while index > 0:
+                self.pop_helper(index)
+                index = index - 1
+
+    def push_helper(self, index):
+        while index // 2 > 0:
+            if self.heapList[index].higher_pri_than(self.heapList[index // 2]):
+                temp = self.heapList[index // 2]
+                self.heapList[index // 2] = self.heapList[index]
+                self.heapList[index] = temp
+            index = index // 2
+
+    # inserts value and rebuilds the heap
+    def push(self, item):
+        self.heapList.append(item)
+        self._size += 1
+        self.push_helper(self._size)
+
+    # removes the smallest element
+    def pop(self):
+        value = self.heapList[1]
+        self.heapList[1] = self.heapList[self._size]
+        self.heapList.pop()
+        self._size -= 1
+        self.pop_helper(1)
+
+        return value
+
+    def peek(self):
+        val = self.heapList[1]
+        return val
+
+    def pop_helper(self, index):
+        leftChildIndex = 2 * index
```

Comment:
This variable name will raise linting errors. Convention is it should be lowercase.